### DISCUSSION TITLE
Are internal links that open in a new tab considered external anchors?

### DISCUSSION THREAD

**Discussion by rosemaryluo** [twilio-labs/paste#1645] [Q&A] [answered]
Hi team, looking for clarification on what we consider an external anchor--I've seen the icon used before for pages that will open in a new tab. Is that behavior is implied by the icon?
Context: We have some internal pages that currently open in new tabs, should we use the external anchor icon?

**Comment by gloriliale** [comment, marked as answer]
Hi Rosemary, thanks for asking!

When the link is for an external page (not in the same domain), our Anchor component opens that in a new tab by default, though that functionality can be overriden. The icon added from the `showExternal` prop is used for indicating you'll be going to an external page.

If you have a link that stays in the same domain (not an external link) but want it to open in a new tab, we'd recommend stating that in/around the anchor, depending on the context it's in. For example, having the anchor say "[Link] (opens in a new tab)". 

Hopefully that distinction helps. Feel free to give more detail about your specific situation if you'd like more support!

**Reply by rosemaryluo** [reply]
Hmmm it's a footer link with several others so I'm not sure I'd want to add "(opens in a new tab)" to all of them. We're not currently on Paste but want to follow where possible to make transition into Paste easier later. Screenshots here: https://code.hq.twilio.com/deved/twilio-docs/pull/3132

**Reply by serifluous** [reply]
Nice, this is super helpful! Since this is in CodeExchange, there are a few things you could consider:

- When a customer is inside and logged into a Twilio product, the external link icon mainly indicates that they're going to be taken out of the "logged in" experience (for security reasons). Since you're working in CodeExchange and it's not a logged-in experience, using the icon to indicate a new tab is probably fine. You could also reason that going to Docs and "Getting a consultation" (I assume this links to a sales page?) are technically pages that are "external" to CodeExchange, so using the icon is fine in that case, too.
- On the flip side, you might also want to consider how necessary it is that customers need to know that the link will open a new tab up front. If they want it to open in a new tab, they might cmd+click the link anyway, and the icon might not be needed.

Either way, there's no hard guideline on how we indicate new tabs in Twilio (yet), especially in experiences that don't require a user to log in. So it's mainly up to customer preference and your expertise here!

**Reply by rosemaryluo** [reply]
Got it, thanks for the guidance!
